### PR TITLE
Add builder for certificate registration requests

### PR DESCRIPTION
**Pull request by supl**
This PR adds a port for building certificate registration requests.

This is the correspondent of https://github.com/scalar-labs/scalardl-javascript-sdk-base/blob/master/request/builder.js#L36

### CODE REVIEW

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by jnmt — APPROVED** [review]
LGTM! Thank you!

File: node/common/builder/certificate_registration_request.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    this.certHolderId = id;
+    return this;
+  }
+
+  /**
+   * @param {number} version
+   * @return {CertificateRegistrationRequestBuilder}
+   */
+  withCertVersion(version: number): CertificateRegistrationRequestBuilder {
+    this.certVersion = version;
```
Comment (feeblefakie) [inline node/common/builder/certificate_registration_request.ts:27]: Can we do a simple validation? (Like disallowing negative numbers)

File: node/common/builder/certificate_registration_request.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    this.certPem = pem;
+    return this;
+  }
+
+  /**
+   * @return {Promise<CertificateRegistrationRequest>}
+   */
+  async build(): Promise<CertificateRegistrationRequest> {
+    const request = this.request;
+    request.setCertHolderId(this.certHolderId);
```
Comment (feeblefakie) [inline node/common/builder/certificate_registration_request.ts:45]: Do we expect all the values are set?
If so, I think we should check and throw an error as needed.

**Review by feeblefakie — COMMENTED** [review]
Left some comments. PTAL!

**Comment by supl** [comment]
@feeblefakie 

Thank you for your comment.

Sorry for my late reply. I thought I responded already.

> Can we do a simple validation? (Like disallowing negative numbers)

> Do we expect all the values are set?
> If so, I think we should check and throw an error as needed.

Regarding the validation, what I am thinking is, that these "builders" are only used internally therefore all required properties will be set correctly by the caller (the implementation of ClientService)

The validation is conducted in the current ClientService implementation so I think it might be a bit redundant in the builder layer.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**